sources/packages/api/src/route-controllers/note/models/note.dto.ts

```diff
@@ -1,16 +1,67 @@
+import { IsIn, IsNotEmpty, MaxLength } from "class-validator";
+import {
+  Note,
+  NOTE_DESCRIPTION_MAX_LENGTH,
+  User,
+} from "../../../database/entities";
+
 /**
  * Base DTO for note.
  */
-export interface NoteBaseDTO {
+export class NoteAPIInDTO {
+  @IsNotEmpty()
   noteType: string;
+  @IsNotEmpty()
+  @IsIn([
+    "General",
+    "Application",
+    "Program",
+    "Restriction",
+    "Designation",
+    "System Actions",
+  ])
+  @MaxLength(NOTE_DESCRIPTION_MAX_LENGTH)
   description: string;
 }
 
 /**
  * Notes detail DTO. This is used for view only purpose.
  */
-export interface NoteDTO extends NoteBaseDTO {
+export class NoteAPIOutDTO {
+  noteType: string;
+  description: string;
   firstName: string;
   lastName: string;
   createdAt: Date;
 }
+
+/**
+ * Util to transform note entity model to DTO.
+ * @param Entity
+ * @returns notes
+ */
+export const transformToNoteDTO = (note: Note): NoteAPIOutDTO => {
```

Comment:
Is this a method used only once, if yes, I would recommend removing it.

Reply:
Its used twice.